eva-server/src/main/java/uk/ac/ebi/eva/server/ws/ga4gh/beaconv2/BeaconFunctionsV2.java

```diff
@@ -0,0 +1,342 @@
+import java.util.HashSet;
+import java.util.Arrays;
+import java.util.Collections;
+
+@Component
+public class BeaconFunctionsV2 extends EvaWSServer {
+
+    @Autowired
+    private VariantWithSamplesAndAnnotationsService service;
+
+    @Autowired
+    private VariantSourceService variantSourceService;
+
+    public BeaconFunctionsV2() {
+    }
+
+    public BeaconImpl rootGet() {
+        MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName("hsapiens_grch37"));
+        BeaconImpl response = new BeaconImpl();
+        List<BeaconDataset> beaconDatasets = new ArrayList<>();
+        List<VariantSource> variantSourceMongos = variantSourceService.findAllVariantSourcesForBeacon();
+        variantSourceMongos.forEach(variantSourceMongo -> {
+            beaconDatasets.add(new BeaconDataset()
+                    .id(variantSourceMongo.getStudyId())
+                    .name(variantSourceMongo.getStudyName())
+                    .description("randomDescription")
+                    .assemblyId("GRCh37")
+                    .createDateTime(variantSourceMongo.getDate() == null ? null : variantSourceMongo.getDate()
+                            .toString())
+                    .updateDateTime("updatedatetime")
+                    .sampleCount(variantSourceMongo.getStats() == null ? null : (long) variantSourceMongo.getStats()
+                            .getSamplesCount())
+                    .variantCount(variantSourceMongo.getStats() == null ? null :
+                            (long) variantSourceMongo.getStats().getVariantsCount())
+                    .externalUrl("externalurl"));
+        });
+
+        MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName("hsapiens_grch38"));
+        variantSourceMongos = variantSourceService.findAllVariantSourcesForBeacon();
+        variantSourceMongos.forEach(variantSourceMongo -> {
+            beaconDatasets.add(new BeaconDataset()
+                    .id(variantSourceMongo.getStudyId())
+                    .name(variantSourceMongo.getStudyName())
+                    .description("randomDescription")
+                    .assemblyId("GRCh38")
+                    .createDateTime(variantSourceMongo.getDate() == null ? null : variantSourceMongo.getDate()
+                            .toString())
+                    .updateDateTime("updatedatetime")
+                    .sampleCount(variantSourceMongo.getStats() == null ? null : (long) variantSourceMongo.getStats()
+                            .getSamplesCount())
+                    .variantCount(variantSourceMongo.getStats() == null ? null :
+                            (long) variantSourceMongo.getStats().getVariantsCount())
+                    .externalUrl("externalurl"));
+        });
+
+        response.setDatasets(beaconDatasets);
+        return response;
+    }
+
+    public ResponseEntity<List<BeaconAlleleResponse>> queryGet(String chromosome, Long start, Long startMin,
```

Comment:
This method is also very long and readability could be improve if you extract some parts into private methods

Reply:
I have extracted some parts and have created 2 new private functions.
Is that okay? Please let me know if it is not :)